server/aesdsocket.c

```diff
@@ -0,0 +1,254 @@
+
+// signal handler function prototype
+void signal_handler(int sig)
+{
+	if(sig==SIGINT)
+	{
+		syslog(LOG_INFO," SIGINT occurred");
+	}
+	else if(sig==SIGTERM)
+	{
+		syslog(LOG_INFO," SIGTERM occurred");
+	}
+	
+	unlink("/var/tmp/aesdsocketdata"); //Deletes a file
+	
+	//Close socket and client connection
+	close(sockfd);
+	close(accept_rc);
+	
+	exit(0); //Exit success 
+}
+
+//Main functions
+int main(int argc, char *argv[])
+{
+        // variable initialization
+        int addr_fd=0; 
+        int bind_rc=0;
+        int listen_rc=0;
+        char buffer[1024];
+        int fd=0;
+        int send_rc=0;
+        int i=0;
+        int bytes_to_write=0;
+        ssize_t rec_fd=1;
+        ssize_t wr_bytes=0;
+        socklen_t size=sizeof(struct sockaddr);
+        bool write_flag=false;
+        int total_packet_size=0;
+
+        openlog(NULL,LOG_PID, LOG_USER); // initialize syslog
+        if((argc>1) && strcmp(argv[1],"-d")==0)// check for daemon mode flag
+        {
+                if(daemon(0,0)==-1)// set to daemon mode
+                {
+                        syslog(LOG_ERR, "daemon mode failed");
+                        exit(1);
+                }
+        }
+
+        if(signal(SIGINT,signal_handler)==SIG_ERR)// handle SIGINT signal
+        {
+                syslog(LOG_ERR,"SIGINT failed");
+                exit(2);
+        }
+        if(signal(SIGTERM,signal_handler)==SIG_ERR)
+        {
+                syslog(LOG_ERR,"SIGTERM failed");// handle SIGTERM signal
+                exit(3);
+        }
+
+        //create socket
+        sockfd=socket(PF_INET, SOCK_STREAM, 0);
+        if(sockfd==-1)
+        {
+                syslog(LOG_ERR, "fail to create socket");
+                perror("socket");
+                exit(1);
+        }
+
+        //Get address
+        hints.ai_flags=AI_PASSIVE;// set socket options for passive socket
+        addr_fd=getaddrinfo(NULL,"9000",&hints,&servinfo);// get address info for the socket
+        if(addr_fd !=0)
+        {
+                syslog(LOG_ERR, "fail to get server address");
+                perror("getaddrinfo");
+                exit(2);
+        }
+
+        //Bind
+        bind_rc=bind(sockfd,servinfo->ai_addr,sizeof(struct sockaddr));// bind the socket to an address
+        if(bind_rc==-1)
+        {
+                freeaddrinfo(servinfo);
+                syslog(LOG_ERR, "fails to bind");
+                perror("bind");
+                exit(3);
+        }
+
+        freeaddrinfo(servinfo);// free memory allocated for servinfo struct
+        fd=open("/var/tmp/aesdsocketdata", O_WRONLY | O_CREAT | O_TRUNC, S_IRWXU | S_IRWXG | S_IRWXO); //To open/create a file and give permissions to all
+        if (fd==-1)
+        {
+                perror("open");
+                exit(7);
+        }
+        close(fd);// close the file to create it
+        while(1)
+        {
+
+                //Listen
+                listen_rc=listen(sockfd,10);
```

Comment:
@malolasimman you shouldn't need `listen` in the wait loop, and I think under some conditions this can cause issues.  See discord thread at https://discord.com/channels/1062535844696707115/1079159844252176425/1079209338855505980